sandbox/modules/identity/src/main/java/org/opensearch/identity/authz/PermissionFactory.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.authz;
+
+/**
+ * This class is used to create Permission instances. The factory can create both standard Permissions which have specific
+ * formatting requirements and legacy permissions which are not checked for validity on creation.
+ */
+public class PermissionFactory {
+
+    private final static String[] INVALID_CHARACTERS = new String[] { ":", "" }; // This is a placeholder for what may want to be banned
+
+    // A placeholder for the different resources which a permission may grant a permission based on
+    private static enum QUALIFIED_PERMISSION_TYPES {
+        CLUSTER("cluster", false),
+        INDICES("indices", true),
+        PLUGIN("plugin", true),
+        EXTENSION("extension", true);
+
+        private final String permissionType;
+        private final boolean patternRequired;
+
+        QUALIFIED_PERMISSION_TYPES(final String permissionType, boolean patternRequired) {
+            this.permissionType = permissionType;
+            this.patternRequired = patternRequired;
+        }
+
+        public String getPermissionType() {
+            return this.permissionType;
+        }
+
+        public boolean isResourcePatternRequired() {
+            return this.patternRequired;
+        }
+
+        public static QUALIFIED_PERMISSION_TYPES matchingType(String instancePermissionType) {
+            for (QUALIFIED_PERMISSION_TYPES type : values()) {
+                if (type.permissionType.equalsIgnoreCase(instancePermissionType)) {
+                    return type;
+                }
+            }
+            return null;
+        }
+    };
+
+    /**
+     * This function creates a standard permission instance. It includes checking that the permission that is being created
+     * is properly formatted.
+     */
+    public Permission createPermission(String permissionString) {
```

Comment:
For a future PR; there is a schema for these new permissions which involves include period delineation action namespace/names and a resource.  Rather than only accept a 'string' which's contract is 'you need to understand how the permission is shaped and I'll throw an exception if you got it wrong, instead you could change the interface to make it easier to get the parts right, e.g. the signature could be `createPermissions(final String namespace, final String resourceName, final String actionVerb, final string resourceFilter)`